### PR TITLE
Adding menu settings to disable teleporting

### PR DESCRIPTION
**Pull request by danteruiz**
This PR allow the option to disable teleporting

Ticket - https://highfidelity.manuscript.com/f/cases/20231/Add-setting-to-disable-teleport

### CODE REVIEW

**Comment by danteruiz** [comment]
Test plan
- Goto settings controller, instead of have two radio button called `Teleport only` and `telelport and walking`
you should have two check boxes called `teleporting` and `walking` and if the walking option is check the flying option should appear like before.
- make sure toggle `teleporting` disables and enables teleporting
- make sure toggling `walking` disables and enables walking

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15873/HighFidelity-Beta-PR14563-deaf3e8.apk

File: interface/src/ui/PreferencesDialog.cpp [hunk 1]
```diff
@@ -238,14 +238,15 @@ void setupPreferences() {
...
-        preference->setHeading("Movement mode");
-        preference->setItems(items);
+        auto getter = [myAvatar]()->bool { return myAvatar->getAllowTeleporting(); };
+        auto setter = [myAvatar](bool value) { myAvatar->setAllowTeleporting(value); };
+        auto preference = new CheckPreference(VR_MOVEMENT, "Teleporting", getter, setter);
+        preferences->addPreference(preference);
+    }
+    {
+        auto getter = [myAvatar]()->bool { return myAvatar->useAdvancedMovementControls(); };
+        auto setter = [myAvatar](bool value) { myAvatar->setUseAdvancedMovementControls(value); };
```
Comment (wayne-chen) [inline interface/src/ui/PreferencesDialog.cpp]: I think these getter/setter functions should mirror the name of the option - maybe `getAllowWalking/setAllowWalking`?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15873/HighFidelity-Beta-PR14563-deaf3e8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15873/HighFidelity-Beta-Interface-PR14563-deaf3e8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15873/HighFidelity-Beta-PR14563-deaf3e8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15873/HighFidelity-Beta-Interface-PR14563-deaf3e8.exe

File: interface/src/ui/PreferencesDialog.cpp (hunk 1 above)
Comment (danteruiz) [reply interface/src/ui/PreferencesDialog.cpp]: I didn't add that function or edit it. 1. thats the name of the scripting api variable. 2. the previous menu option was called `Teleporting and Walking` it is consistent with what was there before.

Comment (danteruiz) [reply interface/src/ui/PreferencesDialog.cpp]: For those reasons I believe we should just keep what we have.

**Comment by qliejahner** [comment]
Currently testing

**Comment by qliejahner** [comment]
PASS

Settings applied and persisted as expected.

**Pull request merged**